aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -13,6 +13,76 @@ namespace at {
 namespace native {
 namespace {
 
+template <
+    typename scalar_t,
+    typename std::enable_if<std::is_same<c10::Half, scalar_t>::value>::type* =
+        nullptr>
+__device__ __forceinline__ void fastSpecializedAtomicAdd(
```

Comment:
I'd move this to some more general file, rather than anonymous namespace, it is not hard to imagine other kernels wanting to use it